src/main/java/net/fabricmc/loom/configuration/providers/forge/ForgeRunTemplate.java

```diff
@@ -110,12 +111,21 @@ public void applyTo(RunConfigSettings settings, ConfigValue.Resolver configValue
 			settings.getEnvironmentVariables().putIfAbsent(key, resolved);
 		});
 
+		final ForgeProvider forgeProvider = settings.getExtension().getForgeProvider();
+
 		// Add MOD_CLASSES, this is something that ForgeGradle does
-		settings.getEnvironmentVariables().computeIfAbsent("MOD_CLASSES", $ -> ConfigValue.of("{source_roots}").resolve(configValueResolver));
+		settings.getEnvironmentVariables().computeIfAbsent("MOD_CLASSES", $ -> {
+			String modClasses = ConfigValue.of("{source_roots}").resolve(configValueResolver);
 
-		final ForgeProvider forgeProvider = settings.getExtension().getForgeProvider();
+			if (File.pathSeparatorChar == ':' && forgeProvider.usesBootstrapDev()) {
+				// bs-dev has a bug where ';' is used instead of File.pathSeparatorChar
+				modClasses = modClasses.replaceAll(":", ";");
```

Comment:
```suggestion
				modClasses = modClasses.replace(":", ";");
```
No need to use regex here